tokio-buf/src/ext/collect.rs

```diff
@@ -29,7 +30,7 @@ enum Error<T, U> {
 
 impl<T, U> Collect<T, U>
 where
-    T: BufStream,
+    T: BufStreamExt,
```

Comment:
I don't think this is actually needed. I *think* you can bound by `T: BufStream` since there is a blanket impl.